src/ml_tooling/utils.py

```diff
@@ -129,16 +138,29 @@ def _create_param_grid(pipe: Pipeline, param_grid: dict) -> ParameterGrid:
     return ParameterGrid(step_dict)
 
 
-def _validate_estimator(model):
+def _validate_estimator(estimator):
     """
-    Ensures that estimator runs
-    :param model:
-    :return:
+    Ensures that estimator is a valid estimator - either a :class:`~sklearn.base.BaseEstimator`
+    or a :class:`~sklearn.pipeline.Pipeline` with a :class:`~sklearn.base.BaseEstimator`
+    as the final step
+
+    Parameters
+    ----------
+    estimator: passed estimator to validate
+
+    Returns
+    -------
+    :class:`~sklearn.base.BaseEstimator`
+
+    Raises
+    ------
+    MLToolingError
+        Raises on invalid input
     """
-    if hasattr(model, '_estimator_type'):
-        return model
+    if hasattr(estimator, '_estimator_type'):
```

Comment:
I think it would be logical to read if the structure was:

```python
if hasattr(estimator, '_estimator_type'):
        return estimator
elif isinstance(estimator, Pipeline):
        raise MLToolingError("You passed a Pipeline without an estimator as the last step")
else:
        raise MLToolingError(f"Expected a Pipeline or Estimator - got {type(estimator)}")
```

Reply:
I would prefer to keep it the way it is written, as it's more pythonic.

In general, return and raise serve as control flow in Python - if an if statement returns, then that is the logical end of the condition, since it is not possible to continue past a return. Keeps indentation lower as well :smile: